apps/web/src/pages/_app.tsx

```diff
@@ -93,7 +98,7 @@ function MyApp(props: AppProps<{ initialReduxState: any; dehydratedState: any }>
           <GlobalCheckClaimStatus excludeLocations={[]} />
           <PersistGate loading={null} persistor={persistor}>
             <Updaters />
-            <App {...props} />
+            <InnerApp {...props} isUserIpWhitelisted={isUserIpWhitelisted} />
```

Comment:
Let's leave the app untouched. We can create a context called `ServerContext` as one of the providers so the components inside the application can access context passed from the server end

Reply:
the name of the flag for notification can be `notificationEnabled`